slatedb/src/db.rs

```diff
@@ -2624,6 +2632,110 @@ mod tests {
         assert!(found_keys.contains(key2.as_slice()));
     }
 
+    #[cfg(feature = "test-util")]
+    async fn test_sequence_tracker_persisted_across_flush_and_reload_impl(wal_enabled: bool) {
```

Comment:
Nit: maybe use proptest instead of the two test functions below?

Reply:
I tried that but it gets a bit weird with the `#[cfg(feature = wal_disable)]`. I could gate the entire proptest on it, but that felt wrong 🤔